File: No_1838/Frequency of the Most Frequent Element.py
class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        nums.sort()
        total = nums[0]
        left = 0
        ret = 1
        for right in range(1, len(nums)):
            total += nums[right]
            while total + k < nums[right] * (right - left + 1):
                total -= nums[left]
                left += 1
            ret = max(ret, right - left + 1)
        
        return ret
        

File: No_1838/Frequency of the Most Frequent Element.cpp
class Solution {
public:
    int maxFrequency(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int ret = 1;
        long long sum = nums[0];
        int left = 0;
        for (int right = 1; right < nums.size(); right++) {
            sum += nums[right];
            while (sum + k < (long long)nums[right] * (right - left + 1)) {
                sum -= nums[left];
                left++;
            }   
            ret = max(ret, right - left + 1);
        }
        
        return ret;
    }
};